### PR TITLE
UI - Navbar for MapKnitter

### PR DESCRIPTION
**Pull request by stefannibrasil**
Related to #327 

Hi all,
My proposal here is to change the navbar to use the same layout as Public Lab. That way, whenever we need to create a new navbar for a new project, we can follow this pattern. This also prevents having different CSS files and classes for every project, letting us focus on the project layout.

I still need to fix some small issues, like the login form and the search form.

But I have some questions that I wanted to ask before moving on:

* The app is not using Bootstrap at all? I couldn't find a link to it, so I added a new one for Bootstrap 4. I may be missing something here, but it does not look like we are using Bootstrap. I am totally for using 
Bootstrap 4, they have some nice cards and mobile utilities that make our work easier.

* I noticed that the Resources link on the navbar was being hidden. Is it supposed to be removed from it?

Let me know your thoughts, please! @jywarren 

Some screenshots: to help with the review:

screen shot 2019-02-25 at 11 57 11
screenshot_2019-02-25 mapknitter

### CODE REVIEW

**Comment by jywarren** [comment]
Hi!!! So, like in plots2, the stylesheets are included here: https://github.com/publiclab/mapknitter/blob/master/app/views/layouts/application.html.erb#L16

bootstrap is included, and we seem to be on Bootstrap 3.2 -- see:

https://github.com/publiclab/mapknitter/blob/master/app/assets/stylesheets/application.css#L12

https://github.com/publiclab/mapknitter/blob/master/bower.json#L6

We should move to Yarn, see https://github.com/publiclab/mapknitter/pull/357 for progress on this, which should simplify things a bit. 

+1 this change, the menubar would be nice if standardized here!

**Comment by stefannibrasil** [comment]
ahhh, thanks!!! So we are using Bootstrap 3.2.0 
https://github.com/publiclab/mapknitter/blob/master/bower.json#L6

I'll update the changes then. I'm not familiar with yarn and bower, but if that will simplify, awesome! :tada

I will try to finish today (tomorrow for sure), then I'll upload more detailed screenshots for this. Thanks for the help, @jywarren

**Comment by jywarren** [comment]
But we should look at upgrading to bootstrap 4!! Shouldn't be too hard
compared to plots2 since this is such a simpler app!

**Comment by stefannibrasil** [comment]
I'm totally for it! Maybe creating an issue for now? After this navbar, I can take a look.

**Comment by jywarren** [comment]
sounds great!

**Comment by stefannibrasil** [comment]
okay, I created this new issue #377  for the Bootstrap update. If you want to add it to projects page and add labels to it, it would be nice, I couldn't add any :+1:

**Comment by stefannibrasil** [comment]
Hi @jywarren I changed the navbar to use Bootstrap 3.2.0 for now.

Here are some screenshots. One thing that I noticed was that the navbar was not using Bootstrap's collapse, so I changed that, no need for adding CSS for this behavior.

**Comment by stefannibrasil** [comment]
screenshot_2019-03-02 mapknitter 2


When I take screenshots the navbar gets in the middle of the map, IDK why!
Here I moved the search form to the navbar, but I left the previous one there for now.

**Comment by stefannibrasil** [comment]
screenshot_2019-03-02 mapknitter

this one looks better to see the navbar alone.

**Comment by stefannibrasil** [comment]
Mobile:


screen shot 2019-03-01 at 17 39 13



Opening the user dropdown:


screen shot 2019-03-01 at 17 39 23

**Comment by stefannibrasil** [comment]
Maps page:

screen shot 2019-03-01 at 17 38 53
screenshot_2019-03-02 mapknitter

**Comment by stefannibrasil** [comment]
Let me know what do you think! It leaves more space for adding contents later, I think :)

**Comment by jywarren** [comment]
Ooh it's really nice! Is it ready for a merge? Thanks!!!

**Comment by jywarren** [comment]
It looks like Travis is OK now?

**Comment by stefannibrasil** [comment]
yeah, it's ready for a merge! Glad you liked :)
I think after I switched to normal PR travis ran as expected, so it's okay now.

After we merge this, should I start on updating Bootstrap version? Thanks :v:

File: app/assets/stylesheets/header.css.scss [hunk 1]
```diff
@@ -1,130 +1 @@
...
-
-  .brand span {
-    display:block;
-    float:left;
-    padding-top:65px;
-    font-size:20px;
-    font-family: 'Junction Light';
-  }
-
-  img.cat {
```
Comment (jywarren) [inline app/assets/stylesheets/header.css.scss]: Did you find the easter egg???????

**Comment by jywarren** [comment]
🎉 !!!! This is awesome. Thanks @stefannibrasil -- yeah that'd be great!

**Pull request merged**